### PR TITLE
PROD-19 Contenful api integration

### PR DESCRIPTION
**Pull request by Valdas-Stonkus**
Created fetch api service to get content array of content objects.

### CODE REVIEW

File: src/api/contenful.mappper.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export class ContentfulResponseMapper {
+    constructor() {
+        this.posts = []
```
Comment (prodevshub) [inline src/api/contenful.mappper.js]: Mapper responsibility is just convert response to the data structure you want, contenful api is the place to cache post and check if there is cache before make a api call. Move this state to api class.

File: src/api/contenful.mappper.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    }
+
+    postsResponseToPostList = (response) => {
+        this.posts = response.fields.posts.map((post) => ({
```
Comment (prodevshub) [inline src/api/contenful.mappper.js]: just return, no storing.

**Review by prodevshub — CHANGES_REQUESTED** [review]

File: src/api/contenful.mappper.js (hunk 1 above)
Comment (Valdas-Stonkus) [reply src/api/contenful.mappper.js]: Done

File: src/api/contenful.mappper.js (hunk 2 above)
Comment (Valdas-Stonkus) [reply src/api/contenful.mappper.js]: Done

**Comment by Valdas-Stonkus** [comment]
Changes made, review please

File: src/api/contentful.api.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { ContentfulResponseMapper } from './contenful.mappper'
+import { CONTENTFUL_SPACE_ID, CONTENTFUL_ACCESS_TOKEN } from '../lib/constants'
+
+const contentful = require('contentful')
+
+class PostsApi {
+    constructor() {
+        this.posts = new ContentfulResponseMapper()
```
Comment (prodevshub) [inline src/api/contentful.api.js]: This name is misleading, it should be named to this.mapper

**Review by prodevshub — CHANGES_REQUESTED** [review]

File: src/api/contentful.api.js (hunk 3 above)
Comment (Valdas-Stonkus) [reply src/api/contentful.api.js]: You wrote early move this.post to api. Where must be this.post then?

**Review by prodevshub — APPROVED** [review]

**Pull request merged**